mage_ai/frontend/pages/pipelines/[pipeline]/edit.tsx

```diff
@@ -1331,6 +1325,46 @@ function PipelineDetailPage({
     onChangeCodeBlock,
   ]);
 
+  // Check for pipeline or project config errors
+  useEffect(() => {
+    let dataWithPotentialError = data;
+    let configFileLinks = [];
+    if (!data?.hasOwnProperty('error') && dataDataProviders?.hasOwnProperty('error')) {
+      dataWithPotentialError = dataDataProviders;
+    } else if ((pipeline?.variables_dir?.includes('.mage_data')
```

Comment:
@wangxiaoyou1993 Is checking the `variables_dir` value for `.mage_data` (the default Mage data directory used if no `variables_dir` value provided in the config) a valid way to check for a missing `variables_dir` property? Jw if there's a scenario where users might actually want to use that folder for their variables.

Reply:
That's the default value for variables_dir so that's a valid value.

Reply:
We just check if the `variables_dir` and `remote_variables_dir` values have `None` in it, which could be caused by using an env var that has an empty or no value. If either of those do, we display an error popup with a link to the project's `metadata.yaml` file.

Leaving the `variables_dir` and `remote_variables_dir` values blank in the `metadata.yaml` config is valid, so no error popup will appear if those values are blank, since the `variables_dir` will just be the default value and the `remote_variables_dir` is not required.